ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/index.test.ts

```diff
@@ -162,4 +168,34 @@ describe("main function export", () => {
     expect(reportExceptionsMock).not.toHaveBeenCalled();
     expect(deleteMessagesMock).not.toHaveBeenCalled();
   });
+
+  it("messages to failure queue response is 400", async () =>{
+    // GIVEN
+    prepareQueue([
+      { messageText: JSON.stringify({ a: "b" }) },
+      { messageText: JSON.stringify({ c: "d" }) },
+    ]);
+
+    uploadResultMock = jest.spyOn(lib, "uploadResult").mockResolvedValue({
+      ok: false,
+      status: 400,
+      json: () => Promise.resolve({ destinationCount: 4 }),
+      text: () => Promise.resolve("error"),
+    } as jest.Mocked<Response>);
+
+    // WHEN
+    try {
+      await fn(context);
+      expect(0).toBe(1);
```

Comment:
what's the purpose of this line?

Reply:
So this is copy pasta from another test so I'm not certain, but I'm pretty sure this is just an assertion to ensure that the function call throws an exception.  i.e. if we reach this line the test should fail

Reply:
I think this might've been left in accidentally or something? because I made it reach this line and the test still passed since `expect(0).toBe(1)` throws its own exception. I think it needs to verify that `e` is a specific type of exception for it to fail

Reply:
weird